src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -195,6 +204,13 @@ create_distributed_table(PG_FUNCTION_ARGS)
 							   colocateWithTableName, ShardCount,
 							   ShardReplicationFactor);
 
+	/* copy over data from regular relations */
+	relationKind = get_rel_relkind(relationId);
+	if (relationKind == RELKIND_RELATION)
+	{
+		CopyLocalData(relationId);
```

Comment:
Is there any chance this can be pushed down into `ConvertToDistributedTable`? I know it's only copy-pasted once right here, but that function already does most of the rest of the distributed table configuration (including getting an exclusive lock), so it seems better to have this part of the "create a table" arc included in that flow. Maybe?